core/src/setup/Kokkos_Setup_HIP.hpp

```diff
@@ -19,6 +19,12 @@
 
 #if defined(KOKKOS_ENABLE_HIP)
 
+#if defined(KOKKOS_ARCH_AMD_GFX942) && defined(KOKKOS_ARCH_AMD_GFX942_APU)
+static_assert(false,
+              "Kokkos detected both `KOKKOS_ARCH_AMD_GFX942` and "
+              "`KOKKOS_ARCH_AMD_GFX942_APU` which is not allowed.");
```

Comment:
I'd say "internal error" but this is good enough

Reply:
I meant as a prefix `Kokkos Internal Error: Conflicting architecture definitions.  Both KOKKOS_ARCH_AMD_GFX942 and KOKKOS_ARCH_AMD_GFX942_APU are defined.  Only one architecture can be specified.`

Anyway I don't think it is worth rerunning the CI